File: web/src/api/scheduled-deliveries.js

import apiService from './apiService'

/**
 * Service pour les livraisons planifiées côté web
 */
class ScheduledDeliveryService {
  constructor() {
    this.baseURL = '/scheduled-deliveries'
  }

  /**
   * Créer une nouvelle livraison planifiée
   */
  async createScheduledDelivery(data) {
    try {
      const response = await apiService.post(this.baseURL, data)
      return response
    } catch (error) {
      console.error('Erreur lors de la création de la livraison planifiée:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la création')
    }
  }

  /**
   * Récupérer les livraisons planifiées
   */
  async getScheduledDeliveries(params = {}) {
    try {
      const queryParams = new URLSearchParams()
      
      if (params.status) queryParams.append('status', params.status)
      if (params.skip !== undefined) queryParams.append('skip', params.skip.toString())
      if (params.limit !== undefined) queryParams.append('limit', params.limit.toString())
      if (params.client_id) queryParams.append('client_id', params.client_id.toString())
      if (params.start_date) queryParams.append('start_date', params.start_date)
      if (params.end_date) queryParams.append('end_date', params.end_date)

      const url = queryParams.toString() ? `${this.baseURL}?${queryParams}` : this.baseURL
      const response = await apiService.get(url)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération des livraisons planifiées:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Récupérer une livraison planifiée par ID
   */
  async getScheduledDelivery(scheduleId) {
    try {
      const response = await apiService.get(`${this.baseURL}/${scheduleId}`)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération de la livraison planifiée:', error)
      throw new Error(error.response?.data?.detail || 'Livraison planifiée non trouvée')
    }
  }

  /**
   * Mettre à jour une livraison planifiée
   */
  async updateScheduledDelivery(scheduleId, data) {
    try {
      const response = await apiService.put(`${this.baseURL}/${scheduleId}`, data)
      return response
    } catch (error) {
      console.error('Erreur lors de la mise à jour de la livraison planifiée:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la mise à jour')
    }
  }

  /**
   * Supprimer une livraison planifiée
   */
  async deleteScheduledDelivery(scheduleId) {
    try {
      const response = await apiService.delete(`${this.baseURL}/${scheduleId}`)
      return response
    } catch (error) {
      console.error('Erreur lors de la suppression de la livraison planifiée:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la suppression')
    }
  }

  /**
   * Mettre en pause une livraison planifiée
   */
  async pauseScheduledDelivery(scheduleId) {
    try {
      const response = await apiService.post(`${this.baseURL}/${scheduleId}/pause`)
      return response
    } catch (error) {
      console.error('Erreur lors de la mise en pause:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la mise en pause')
    }
  }

  /**
   * Reprendre une livraison planifiée
   */
  async resumeScheduledDelivery(scheduleId) {
    try {
      const response = await apiService.post(`${this.baseURL}/${scheduleId}/resume`)
      return response
    } catch (error) {
      console.error('Erreur lors de la reprise:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la reprise')
    }
  }

  /**
   * Exécuter manuellement une livraison planifiée
   */
  async executeScheduledDelivery(scheduleId) {
    try {
      const response = await apiService.post(`${this.baseURL}/${scheduleId}/execute`)
      return response
    } catch (error) {
      console.error('Erreur lors de l\'exécution:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de l\'exécution')
    }
  }

  /**
   * Récupérer les événements du calendrier
   */
  async getCalendarEvents(startDate, endDate) {
    try {
      const params = new URLSearchParams()
      params.append('start_date', startDate)
      params.append('end_date', endDate)

      const response = await apiService.get(`${this.baseURL}/calendar/events?${params}`)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération des événements:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Créer plusieurs livraisons planifiées en une fois
   */
  async bulkCreateScheduledDeliveries(data) {
    try {
      const response = await apiService.post(`${this.baseURL}/bulk-create`, data)
      return response
    } catch (error) {
      console.error('Erreur lors de la création en lot:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la création en lot')
    }
  }

  /**
   * Récupérer les statistiques des livraisons planifiées
   */
  async getScheduledDeliveryStats() {
    try {
      const response = await apiService.get(`${this.baseURL}/stats/summary`)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération des statistiques:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Coordonner une livraison planifiée (J-1)
   */
  async coordinateScheduledDelivery(executionId, coordinationData) {
    try {
      const response = await apiService.post(`${this.baseURL}/${executionId}/coordinate`, coordinationData)
      return response
    } catch (error) {
      console.error('Erreur lors de la coordination:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la coordination')
    }
  }

  /**
   * Récupérer le statut de coordination
   */
  async getCoordinationStatus(executionId) {
    try {
      const response = await apiService.get(`${this.baseURL}/${executionId}/coordination-status`)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération du statut:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Récupérer les négociations en attente
   */
  async getPendingNegotiations() {
    try {
      const response = await apiService.get('/negotiations/pending')
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération des négociations:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Répondre à une négociation
   */
  async respondToNegotiation(negotiationId, accept, message) {
    try {
      const response = await apiService.post(`/negotiations/${negotiationId}/respond`, {
        accept: accept,
        message: message
      })
      return response
    } catch (error) {
      console.error('Erreur lors de la réponse à la négociation:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la réponse')
    }
  }

  /**
   * Faire une contre-offre
   */
  async createCounterOffer(negotiationId, proposedPrice, message) {
    try {
      const response = await apiService.post(`/negotiations/${negotiationId}/counter-offer`, {
        proposed_price: proposedPrice,
        message: message
      })
      return response
    } catch (error) {
      console.error('Erreur lors de la contre-offre:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la contre-offre')
    }
  }

  /**
   * Récupérer l'historique des négociations
   */
  async getNegotiationHistory(scheduleId) {
    try {
      const response = await apiService.get(`/negotiations/history/${scheduleId}`)
      return response
    } catch (error) {
      console.error('Erreur lors de la récupération de l\'historique:', error)
      throw new Error(error.response?.data?.detail || 'Erreur lors de la récupération')
    }
  }

  /**
   * Valider les données avant envoi
   */
  validateScheduleData(data) {
    const errors = []

    if (!data.title || data.title.trim() === '') {
      errors.push('Le titre est obligatoire')
    }

    if (!data.pickup_address || data.pickup_address.trim() === '') {
      errors.push('L\'adresse de ramassage est obligatoire')
    }

    if (!data.delivery_address || data.delivery_address.trim() === '') {
      errors.push('L\'adresse de livraison est obligatoire')
    }

    if (!data.scheduled_date) {
      errors.push('La date de planification est obligatoire')
    }

    if (data.proposed_price && (isNaN(data.proposed_price) || data.proposed_price <= 0)) {
      errors.push('Le prix proposé doit être un nombre positif')
    }

    if (data.recurrence_type && !['none', 'daily', 'weekly', 'monthly'].includes(data.recurrence_type)) {
      errors.push('Type de récurrence invalide')
    }

    return errors
  }

  /**
   * Formatter les données pour l'API
   */
  formatScheduleDataForAPI(formData) {
    const apiData = {
      title: formData.title,
      description: formData.description || '',
      pickup_address: formData.pickup_address,
      pickup_commune: formData.pickup_commune || '',
      pickup_lat: formData.pickup_lat || null,
      pickup_lng: formData.pickup_lng || null,
      pickup_contact_name: formData.pickup_contact_name || '',
      pickup_contact_phone: formData.pickup_contact_phone || '',
      pickup_instructions: formData.pickup_instructions || '',
      delivery_address: formData.delivery_address,
      delivery_commune: formData.delivery_commune || '',
      delivery_lat: formData.delivery_lat || null,
      delivery_lng: formData.delivery_lng || null,
      delivery_contact_name: formData.delivery_contact_name || '',
      delivery_contact_phone: formData.delivery_contact_phone || '',
      delivery_instructions: formData.delivery_instructions || '',
      package_description: formData.package_description || '',
      package_size: formData.package_size || 'medium',
      package_weight: formData.package_weight ? parseFloat(formData.package_weight) : null,
      is_fragile: Boolean(formData.is_fragile),
      cargo_category: formData.cargo_category || '',
      required_vehicle_type: formData.required_vehicle_type || '',
      proposed_price: formData.proposed_price ? parseFloat(formData.proposed_price) : null,
      delivery_type: formData.delivery_type || 'standard',
      special_instructions: formData.special_instructions || '',
      scheduled_date: formData.scheduled_date,
      recurrence_type: formData.recurrence_type || 'none',
      recurrence_interval: formData.recurrence_interval ? parseInt(formData.recurrence_interval) : null,
      recurrence_days: formData.recurrence_days || [],
      end_date: formData.end_date || null,
      max_occurrences: formData.max_occurrences ? parseInt(formData.max_occurrences) : null,
      notification_advance_hours: formData.notification_advance_hours ? parseInt(formData.notification_advance_hours) : 24,
      auto_create_delivery: Boolean(formData.auto_create_delivery)
    }

    return apiData
  }
}

export default new ScheduledDeliveryService()

File: backend/app/api/scheduled_deliveries.py

from fastapi import APIRouter, Depends, HTTPException, Query, BackgroundTasks
from sqlalchemy.orm import Session
from typing import List, Optional
from datetime import datetime, timedelta

from ..db.session import get_db
from ..core.security import get_current_user
from ..models.user import User
from ..schemas import scheduled_delivery as scheduled_schemas
from ..services.scheduled_delivery import ScheduledDeliveryService
import logging

logger = logging.getLogger(__name__)

router = APIRouter()

@router.post("/scheduled-deliveries", response_model=scheduled_schemas.ScheduledDeliveryResponse)
async def create_scheduled_delivery(
    schedule_data: scheduled_schemas.ScheduledDeliveryCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Créer une nouvelle livraison planifiée"""
    
    if current_user.role not in ["client", "business"]:
        raise HTTPException(status_code=403, detail="Seuls les clients et entreprises peuvent planifier des livraisons")
    
    try:
        schedule = ScheduledDeliveryService.create_scheduled_delivery(db, schedule_data, current_user.id)
        
        return {
            "id": schedule.id,
            "client_id": schedule.client_id,
            "title": schedule.title,
            "description": schedule.description,
            "pickup_address": schedule.pickup_address,
            "pickup_commune": schedule.pickup_commune,
            "pickup_lat": schedule.pickup_lat,
            "pickup_lng": schedule.pickup_lng,
            "pickup_contact_name": schedule.pickup_contact_name,
            "pickup_contact_phone": schedule.pickup_contact_phone,
            "pickup_instructions": schedule.pickup_instructions,
            "delivery_address": schedule.delivery_address,
            "delivery_commune": schedule.delivery_commune,
            "delivery_lat": schedule.delivery_lat,
            "delivery_lng": schedule.delivery_lng,
            "delivery_contact_name": schedule.delivery_contact_name,
            "delivery_contact_phone": schedule.delivery_contact_phone,
            "delivery_instructions": schedule.delivery_instructions,
            "package_description": schedule.package_description,
            "package_size": schedule.package_size,
            "package_weight": schedule.package_weight,
            "is_fragile": schedule.is_fragile,
            "cargo_category": schedule.cargo_category,
            "required_vehicle_type": schedule.required_vehicle_type,
            "proposed_price": schedule.proposed_price,
            "delivery_type": schedule.delivery_type,
            "special_instructions": schedule.special_instructions,
            "scheduled_date": schedule.scheduled_date,
            "recurrence_type": schedule.recurrence_type,
            "recurrence_interval": schedule.recurrence_interval,
            "recurrence_days": schedule.recurrence_days,
            "end_date": schedule.end_date,
            "max_occurrences": schedule.max_occurrences,
            "notification_advance_hours": schedule.notification_advance_hours,
            "auto_create_delivery": schedule.auto_create_delivery,
            "status": schedule.status,
            "created_at": schedule.created_at,
            "updated_at": schedule.updated_at,
            "last_executed_at": schedule.last_executed_at,
            "next_execution_at": schedule.next_execution_at,
            "total_executions": schedule.total_executions,
            "client": {
                "id": current_user.id,
                "name": current_user.full_name,
                "email": current_user.email,
                "phone": current_user.phone
            }
        }
        
    except Exception as e:
        logger.error(f"Erreur création livraison planifiée: {str(e)}", exc_info=True)
        raise HTTPException(status_code=400, detail=f"Erreur lors de la création: {str(e)}")

@router.get("/scheduled-deliveries")
async def get_scheduled_deliveries(
    status: Optional[str] = Query(None, description="Statut des planifications"),
    skip: int = Query(0, ge=0, description="Nombre d'éléments à ignorer"),
    limit: int = Query(20, ge=1, le=100, description="Nombre maximum d'éléments"),
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Récupérer les livraisons planifiées de l'utilisateur"""
    
    try:
        # Pour les clients, filtrer par leur ID
        client_id = None if current_user.role in ["admin", "manager"] else current_user.id
        
        schedules = ScheduledDeliveryService.get_scheduled_deliveries(
            db, client_id=client_id, status=status, skip=skip, limit=limit
        )
        
        return {
            "success": True,
            "schedules": schedules,
            "total": len(schedules),
            "skip": skip,
            "limit": limit
        }
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la récupération: {str(e)}")

@router.get("/scheduled-deliveries/{schedule_id}")
async def get_scheduled_delivery(
    schedule_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Récupérer une livraison planifiée par son ID"""
    
    schedule = ScheduledDeliveryService.get_scheduled_delivery(db, schedule_id)
    if not schedule:
        raise HTTPException(status_code=404, detail="Planification non trouvée")
    
    # Vérifier les permissions
    if current_user.role not in ["admin", "manager"] and schedule.client_id != current_user.id:
        raise HTTPException(status_code=403, detail="Accès non autorisé")
    
    return {"success": True, "schedule": schedule}

@router.put("/scheduled-deliveries/{schedule_id}")
async def update_scheduled_delivery(
    schedule_id: int,
    update_data: scheduled_schemas.ScheduledDeliveryUpdate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Mettre à jour une livraison planifiée"""
    
    schedule = ScheduledDeliveryService.get_scheduled_delivery(db, schedule_id)
    if not schedule:
        raise HTTPException(status_code=404, detail="Planification non trouvée")
    
    # Vérifier les permissions
    if current_user.role not in ["admin", "manager"] and schedule.client_id != current_user.id:
        raise HTTPException(status_code=403, detail="Accès non autorisé")
    
    try:
        updated_schedule = ScheduledDeliveryService.update_scheduled_delivery(db, schedule_id, update_data)
        return {"success": True, "schedule": updated_schedule}
        
    except Exception as e:
        raise HTTPException(status_code=400, detail=f"Erreur lors de la mise à jour: {str(e)}")

@router.delete("/scheduled-deliveries/{schedule_id}")
async def delete_scheduled_delivery(
    schedule_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Supprimer une livraison planifiée"""
    
    schedule = ScheduledDeliveryService.get_scheduled_delivery(db, schedule_id)
    if not schedule:
        raise HTTPException(status_code=404, detail="Planification non trouvée")
    
    # Vérifier les permissions
    if current_user.role not in ["admin", "manager"] and schedule.client_id != current_user.id:
        raise HTTPException(status_code=403, detail="Accès non autorisé")
    
    success = ScheduledDeliveryService.delete_scheduled_delivery(db, schedule_id)
    if success:
        return {"success": True, "message": "Planification supprimée avec succès"}
    else:
        raise HTTPException(status_code=500, detail="Erreur lors de la suppression")

@router.get("/scheduled-deliveries/calendar/events")
async def get_calendar_events(
    start_date: datetime = Query(..., description="Date de début"),
    end_date: datetime = Query(..., description="Date de fin"),
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Récupérer les événements pour le calendrier"""
    
    try:
        # Pour les clients, filtrer par leur ID
        client_id = None if current_user.role in ["admin", "manager"] else current_user.id
        
        events = ScheduledDeliveryService.get_calendar_events(db, start_date, end_date, client_id)
        
        return {"success": True, "events": events}
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la récupération des événements: {str(e)}")

@router.post("/scheduled-deliveries/{schedule_id}/execute")
async def execute_scheduled_delivery(
    schedule_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Exécuter manuellement une livraison planifiée"""
    
    if current_user.role not in ["admin", "manager"]:
        raise HTTPException(status_code=403, detail="Seuls les administrateurs peuvent exécuter manuellement")
    
    # Trouver la prochaine exécution en attente
    from ..models.scheduled_delivery import ScheduledDeliveryExecution
    execution = db.query(ScheduledDeliveryExecution).filter(
        ScheduledDeliveryExecution.scheduled_delivery_id == schedule_id,
        ScheduledDeliveryExecution.status == "pending"
    ).first()
    
    if not execution:
        raise HTTPException(status_code=404, detail="Aucune exécution en attente trouvée")
    
    delivery = ScheduledDeliveryService.execute_scheduled_delivery(db, execution.id)
    
    if delivery:
        return {"success": True, "delivery_id": delivery.id, "message": "Livraison créée avec succès"}
    else:
        raise HTTPException(status_code=500, detail="Erreur lors de l'exécution")

@router.post("/scheduled-deliveries/{schedule_id}/pause")
async def pause_scheduled_delivery(
    schedule_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Mettre en pause une livraison planifiée"""
    
    schedule = ScheduledDeliveryService.get_scheduled_delivery(db, schedule_id)
    if not schedule:
        raise HTTPException(status_code=404, detail="Planification non trouvée")
    
    # Vérifier les permissions
    if current_user.role not in ["admin", "manager"] and schedule.client_id != current_user.id:
        raise HTTPException(status_code=403, detail="Accès non autorisé")
    
    from ..schemas.scheduled_delivery import ScheduledDeliveryUpdate, ScheduledDeliveryStatus
    update_data = ScheduledDeliveryUpdate(status=ScheduledDeliveryStatus.paused)
    
    updated_schedule = ScheduledDeliveryService.update_scheduled_delivery(db, schedule_id, update_data)
    return {"success": True, "schedule": updated_schedule}

@router.post("/scheduled-deliveries/{schedule_id}/resume")
async def resume_scheduled_delivery(
    schedule_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Reprendre une livraison planifiée en pause"""
    
    schedule = ScheduledDeliveryService.get_scheduled_delivery(db, schedule_id)
    if not schedule:
        raise HTTPException(status_code=404, detail="Planification non trouvée")
    
    # Vérifier les permissions
    if current_user.role not in ["admin", "manager"] and schedule.client_id != current_user.id:
        raise HTTPException(status_code=403, detail="Accès non autorisé")
    
    from ..schemas.scheduled_delivery import ScheduledDeliveryUpdate, ScheduledDeliveryStatus
    update_data = ScheduledDeliveryUpdate(status=ScheduledDeliveryStatus.active)
    
    updated_schedule = ScheduledDeliveryService.update_scheduled_delivery(db, schedule_id, update_data)
    return {"success": True, "schedule": updated_schedule}

@router.post("/scheduled-deliveries/bulk-create")
async def bulk_create_scheduled_deliveries(
    bulk_data: scheduled_schemas.BulkScheduleCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Créer plusieurs livraisons planifiées en une fois"""
    
    if current_user.role not in ["client", "business"]:
        raise HTTPException(status_code=403, detail="Seuls les clients et entreprises peuvent planifier des livraisons")
    
    created_schedules = []
    errors = []
    
    for i, schedule_data in enumerate(bulk_data.schedules):
        try:
            # Appliquer les paramètres globaux si fournis
            if bulk_data.apply_to_all:
                for key, value in bulk_data.apply_to_all.items():
                    if hasattr(schedule_data, key) and getattr(schedule_data, key) is None:
                        setattr(schedule_data, key, value)
            
            schedule = ScheduledDeliveryService.create_scheduled_delivery(db, schedule_data, current_user.id)
            created_schedules.append(schedule)
            
        except Exception as e:
            errors.append({"index": i, "error": str(e)})
    
    return {
        "success": True,
        "created_count": len(created_schedules),
        "schedules": created_schedules,
        "errors": errors
    }

@router.get("/scheduled-deliveries/stats/summary")
async def get_scheduled_delivery_stats(
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Récupérer les statistiques des livraisons planifiées"""
    
    try:
        # Pour les clients, filtrer par leur ID
        client_id = None if current_user.role in ["admin", "manager"] else current_user.id
        
        from ..models.scheduled_delivery import ScheduledDelivery, ScheduledDeliveryExecution
        
        # Total planifié
        total_query = db.query(ScheduledDelivery)
        if client_id:
            total_query = total_query.filter(ScheduledDelivery.client_id == client_id)
        total_scheduled = total_query.count()
        
        # Actives
        active_schedules = total_query.filter(ScheduledDelivery.status == "active").count()
        
        # En pause
        paused_schedules = total_query.filter(ScheduledDelivery.status == "paused").count()
        
        # Exécutions ce mois
        from datetime import date
        start_of_month = date.today().replace(day=1)
        executions_this_month = db.query(ScheduledDeliveryExecution).join(ScheduledDelivery).filter(
            ScheduledDeliveryExecution.executed_date >= start_of_month,
            ScheduledDelivery.client_id == client_id if client_id else True
        ).count()
        
        # Exécutions aujourd'hui
        today = date.today()
        executions_today = db.query(ScheduledDeliveryExecution).join(ScheduledDelivery).filter(
            ScheduledDeliveryExecution.planned_date >= today,
            ScheduledDeliveryExecution.planned_date < today + timedelta(days=1),
            ScheduledDelivery.client_id == client_id if client_id else True
        ).count()
        
        # Exécutions cette semaine
        start_of_week = today - timedelta(days=today.weekday())
        end_of_week = start_of_week + timedelta(days=7)
        executions_this_week = db.query(ScheduledDeliveryExecution).join(ScheduledDelivery).filter(
            ScheduledDeliveryExecution.planned_date >= start_of_week,
            ScheduledDeliveryExecution.planned_date < end_of_week,
            ScheduledDelivery.client_id == client_id if client_id else True
        ).count()
        
        stats = {
            "total_scheduled": total_scheduled,
            "active_schedules": active_schedules,
            "paused_schedules": paused_schedules,
            "total_executions_this_month": executions_this_month,
            "upcoming_executions_today": executions_today,
            "upcoming_executions_this_week": executions_this_week,
            "success_rate": 95.0,  # À calculer réellement
            "most_common_recurrence": "weekly"  # À calculer réellement
        }
        
        return {"success": True, "stats": stats}
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la récupération des statistiques: {str(e)}")

# Tâche en arrière-plan pour les notifications
@router.post("/scheduled-deliveries/send-notifications")
async def send_scheduled_notifications(
    background_tasks: BackgroundTasks,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Envoyer les notifications pour les livraisons planifiées (admin seulement)"""
    
    if current_user.role not in ["admin", "manager"]:
        raise HTTPException(status_code=403, detail="Seuls les administrateurs peuvent déclencher les notifications")
    
    background_tasks.add_task(ScheduledDeliveryService.send_execution_notifications, db)
    return {"success": True, "message": "Notifications en cours d'envoi"}

@router.post("/scheduled-deliveries/auto-execute")
async def auto_execute_scheduled_deliveries(
    background_tasks: BackgroundTasks,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Auto-exécuter les livraisons planifiées (admin seulement)"""
    
    if current_user.role not in ["admin", "manager"]:
        raise HTTPException(status_code=403, detail="Seuls les administrateurs peuvent déclencher l'auto-exécution")
    
    background_tasks.add_task(ScheduledDeliveryService.auto_execute_deliveries, db)
    return {"success": True, "message": "Auto-exécution en cours"}
